### PR TITLE
[Do Not Merge][RGW-deployment] - skeleton for rpm build process for cortx-rgw-integration

### PR DESCRIPTION
**Pull request by sachitanands**
skeleton for rpm build process for cortx-rgw-integration. 
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: rgw.conf.sample [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "install_path": "/opt/seagate/cortx",
+  "log_dir": "/var/log",
```
Comment (ujjwalpl) [inline rgw.conf.sample:3]: Remove log_dir and log_level. These are not required as we dont have running service.

File: src/setup/rgw.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#! /usr/bin/python3
```
Comment (ujjwalpl) [inline src/setup/rgw.py:1]: Why is this change ? /bin/env is more standard way to access python as it supports virtual env.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: src/setup/rgw.py (hunk 2 above)
Comment (sachitanands) [reply src/setup/rgw.py:1]: rpm building was failing while compiling these python files.

Comment (sachitanands) [reply src/setup/rgw.py:1]: rpm building was failing while compiling these python files.

File: jenkins/build.sh [hunk 3]
```diff
@@ -59,8 +59,57 @@ rpm -q rpm-build > /dev/null || {
...
+   echo \"Please install above python packages\"
+   echo \"===============================================\"
+fi
+exit \$rc " >> rgw-pre-install
+/bin/chmod +x rgw-pre-install
+
+
+# Building rpm using setuptool utility
+
+/usr/bin/python3 ./setup.py bdist_rpm --release="$REL" --requires python36 \
```
Comment (cdeshmukh) [inline jenkins/build.sh:105]: what is this for ? why building rpm, any required rpm supposed to be part of image

File: rgw-post-uninstall [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (cdeshmukh) [inline rgw-post-uninstall:1]: this is unusual file, in what case we will need to uninstall the rpm anyways. 
In update case also, new container would be just booted with new image

File: rgw.conf.sample (hunk 1 above)
Comment (cdeshmukh) [reply rgw.conf.sample:3]: shouldn't we capture log configs and apply in ceph.conf later ?

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import glob
+from setuptools import setup, find_packages
+import json
+import sys
+from fnmatch import fnmatch
+
+if not os.path.isfile("./rgw.conf.sample"):
```
Comment (cdeshmukh) [inline setup.py:24]: what is this conf.sample for? I think you confirmed in morning meet, concluding that we dont need it

**Comment by sachitanands** [comment]
will push latest changes with new PR and will address all comments mentioned in this PR .

**Pull request closed without merging**